### PR TITLE
Improve documents

### PR DESCRIPTION
**Pull request by nd-02110114**
## Motivation

The current documents are not readable. I think there are some problems.

First, the index.rst is too fat. I think the simpler the index pages is, the more people reads.  Actually, the index page of many OSS is very simple.

Second, the TOC Tree is not good. The point is the order and the content. 
About the order, the last TOC tree is `the contribution guide`, so it is difficult to recognize  the `contribution guide` section. And then, I don't understand why the top of first TOC tree is `Tutorial`. We should put on the `Installation` at the top. Before trying to do tutorials, users basically need to install the package. 
About the contents, some TOC Tree contents like Datasets, DataLoaders and Data classes are hard to recognize the difference for newcomers. Basically, the TOC Tree about API reference is too fat. 

Third, it doesn't have local summaries (like  https://lifesci.dgl.ai/api/utils.mols.html) or local contents (https://lifesci.dgl.ai/api/utils.splitters.html) through all API references. These are really helpful when users search the API which they want.

And, the all docs files are putted on `docs`. It is not easy to find the file we want to add docs in. Especially, **requirements.txt** and **requirements.rst** are really troublesome. Considering about increasing the documents in the future, I think we should reorganize the document directory. 

You may feel these changes is trivial, but I believe it is necessary to repeat these these changes for a good document

## What I did

- Reorganize directory structure
  - the current directory structure is the same as the standard structures like DGL-Life-sci or PyG.
  - all rst files are moved to `docs/source`
  - `docs/source` has `get_started`, `development_guide` and `api_reference` directories and files are moved to the appropriate directory.
- Update conf.py
  - Fix the configuration based on JAXChem and PyG
- Update documents except API reference
  - I add the many comments in this PR.
  
After merging this PR, I will make the PR about updating the API reference docs step by step.
I already have updated about `data`, `splitters`, `transformers` in this branch, https://github.com/nd-02110114/deepchem/tree/fix-api-docs-1

### CODE REVIEW

**Review by rbharath — COMMENTED** [review]
Lots of changes here :). Let me know when this is ready for a review! I did a brief pass through, but it would be useful to have a top level listing of the changes this PR makes and the motivation for them

File: README.md
Comment (nd-02110114) [inline README.md:33]: DeepChem officially tested only Python 3.6 and 3.7, so I fixed.

File: docs/Makefile
Comment (nd-02110114) [inline docs/Makefile:8]: I changed the entry directories.  `source` is most standard in many OSS.

File: docs/requirements.txt
Comment (nd-02110114) [inline docs/requirements.txt:6]: Currently, the GDBTModel doesn't depend on the xgboost package, so I removed.

File: docs/source/development_guide/licence.rst
Comment (nd-02110114) [inline docs/source/development_guide/licence.rst:1]: This content is moved from the original index.rst

File: docs/source/get_started/examples.rst
Comment (nd-02110114) [inline docs/source/get_started/examples.rst:1]: The main modification is to fix the cross-reference links like `load_sampl` or `MultitaskRegressor`

File: docs/source/get_started/tutorial.rst
Comment (nd-02110114) [inline docs/source/get_started/tutorial.rst:1]: I seemed the previous tutorials is under construction. 
I updates the contents by putting on the notebook links.

File: docs/source/index.rst
Comment (nd-02110114) [inline docs/source/index.rst:1]: I modified two things.

- Make index.rst simpler (I think the simpler the index pages is, the more people reads)
  - Move license to `development_guide` section
  - Merge `Getting Involved` to ` About US` section
- Update the outline in the bottom of this file

**Comment by nd-02110114** [comment]
@rbharath 
This is ready for reviews!

File: docs/source/conf.py
Comment (nd-02110114) [inline docs/source/conf.py:41]: - `autosummary_generate` is not used in docs.
- `autodoc_default_flags ` is the deprecated api.
- `numpydoc_class_members_toctree ` is not needed because we don't use `numpydoc` extension.

File: README.md
Comment (rbharath) [reply README.md:33]: This isn't related to this PR, but I think with the latest RDKit DeepChem also works for 3.8. We need to add that into the test suite explicitly though

File: docs/Makefile
Comment (rbharath) [reply docs/Makefile:8]: Good cleanup!

File: docs/source/conf.py
Comment (rbharath) [inline docs/source/conf.py:28]: This is nice! Makes it less buggy with an automatic update

File: docs/source/conf.py
Comment (rbharath) [reply docs/source/conf.py:41]: Do we need `numpydock_class_members_toctree` to handle the fact that our docstrings are written in numpy style? If we don't need it, good to remove!

File: docs/source/conf.py
Comment (rbharath) [inline docs/source/conf.py:109]: This looks cleaner than the existing line linker :)

File: docs/source/index.rst
Comment (rbharath) [inline docs/source/index.rst:60]: Perhaps we should change the top level to the nightly build for now since the conda release is still at 2.3.0

File: docs/tutorial.rst
Comment (rbharath) [inline docs/tutorial.rst:47]: I think it's still useful to have a basic tutorial in the docs itself. Colab is a little more heavy weight and it can be nice to give people a very quick glimpse of DeepChem code. Let's add this back into the new tutorial file. 

Ideally, we could expand this out so it's a self-contained micro-overview of all of DeepChem :)

**Review by rbharath — COMMENTED** [review]
Did a first pass! I like the refactor in general. I have a few suggested tweaks

File: docs/source/conf.py
Comment (nd-02110114) [reply docs/source/conf.py:41]: `numpydock_class_members_toctree`  is only needed by using `numpydoc` extension. But, the current docs use `sphinx.ext.napoleon` for parsing Numpy style docstring instead of `numpydoc`. So, I think this option is not valid, not necessary.

File: docs/source/index.rst
Comment (nd-02110114) [reply docs/source/index.rst:60]: I agreed! I'm fixing!

File: docs/tutorial.rst
Comment (nd-02110114) [reply docs/tutorial.rst:47]: >  it can be nice to give people a very quick glimpse of DeepChem code

I think so, too! I will go back the dataset tutorials and add `featurizer`, `splitter` and `model` sections.

**Comment by nd-02110114** [comment]
This is ready for a second review!

File: docs/source/get_started/installation.rst
Comment (rbharath) [inline docs/source/get_started/installation.rst:170]: I think the copy paste here added in your user id which should be removed

File: docs/source/get_started/tutorial.rst
Comment (rbharath) [reply docs/source/get_started/tutorial.rst:1]: This expanded doctest tutorial is great! Hopefully it will provide people a nice inline code sample example for them to work with :)

**Review by rbharath — APPROVED** [review]
The new tutorial looks really nice! I think this is almost ready to merge in. Have a minor comment in one link where a user id got pasted in but once that's fixed feel free to merge in

File: docs/source/get_started/installation.rst
Comment (nd-02110114) [reply docs/source/get_started/installation.rst]: Yes, this is a mistake. I fixed.

**Comment by nd-02110114** [comment]
I fixed the links and the CI passed. So, I merge in.

**Pull request merged**
